safehttp/mux.go

```diff
@@ -71,18 +71,48 @@ type ServeMux struct {
 	mux *http.ServeMux
 }
 
-func registerHandlers(mux *http.ServeMux, handlers map[string]map[string]handlerConfig) {
-	for pattern, handlersPerMethod := range handlers {
-		handlersPerMethod := handlersPerMethod
-		mux.HandleFunc(pattern, func(w http.ResponseWriter, r *http.Request) {
-			cfg, ok := handlersPerMethod[r.Method]
-			if !ok {
-				http.Error(w, http.StatusText(http.StatusMethodNotAllowed), http.StatusMethodNotAllowed)
-				return
-			}
-			processRequest(cfg, w, r)
-		})
+func registerHandlers(mux *http.ServeMux, handlers map[string]map[string]*handlerConfig) {
+	for pattern, handlerConfigPerMethod := range handlers {
+		httpHandler := stdHandler(handlerConfigPerMethod)
+		mux.Handle(pattern, httpHandler)
+	}
+}
+
+// stdHandler creates an http.Handler using a handlerConfig.
+// The handlerConfig map is needed because http.Handle handles requests for any HTTP request method,
+// which we want to avoid in safehttp.
+func stdHandler(handlerConfigPerMethod map[string]*handlerConfig) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		cfg, ok := handlerConfigPerMethod[r.Method]
+		if !ok {
+			http.Error(w, http.StatusText(http.StatusMethodNotAllowed), http.StatusMethodNotAllowed)
+			return
+		}
+		processRequest(cfg, w, r)
+	})
+}
+
+// TODO rename...
+type HandlerRegistration struct {
+	Method  string
```

Comment:
I'd treat `Method == ""` as "all methods" and do a sanity check inside the `StandardHandler` call to make sure methods don't overlap, the length of the `hr` slice is 1 if `Method == ""` is used and so on.